File: C++/Stacks.cpp
#include <iostream>

class Stack{
    private:
        int top;
        std::string stackArray[100];

    public:
        Stack(){
            int top = -1;
        }

        void push(std::string value){
            if(!isFull()){
                stackArray[++top] = value;
            }else{
                std::cout << "Stack is full!";
            }
        }

        std::string pop(){
            std::string returningValue = stackArray[top--];
            return returningValue;
        }

        int count(){
            return top;
        }

        std::string peek(){
            return stackArray[top];
        }

        void clear(){
            for(int i=0; i<=top-1; i++){
                stackArray[i] = "";
            }
            top = -1;
        }

        bool isFull(){ 
            if(top == 5){
                return true;
            }else{
                return false;
            }
        }
};

int main(){

    Stack stackObj;
    stackObj.push("Arjun");
    stackObj.push("Tom");

    std::cout << stackObj.peek() << std::endl;

    stackObj.push("David");
    stackObj.push("Jim");
    
    std::cout << stackObj.peek() << std::endl;
    std::cout << stackObj.count() << std::endl;
    
    stackObj.pop();
    
    std::cout << stackObj.peek() << std::endl;
    std::cout << stackObj.count() << std::endl;

    stackObj.push("Jim");
    stackObj.push("Eric");
    stackObj.push("Terrance");   
    
    return 0;
}

File: Java/Stacks.java
class Stack{
    private static String[] stackArry = new String[100];
    private int top;

    Stack(){
        top = -1;
    }

    boolean isFull(){
        if(top == 4){
            return true;
        }else{
            return false;
        }
    }
    
    public void push(String value){
        if(!isFull()){
            stackArry[++top] = value;
        }else{
            System.out.println("Stack is Full!!");
        }
    }

    public String pop(){
        String retValue = stackArry[top--];
        return retValue;
    }

    public String peek(){
        return stackArry[top];
    }

    public int count(){
        return top+1;
    }

    public void clear(){
        for(int i=0; i<=top-1; i++){
            stackArry[i] = "";
        }
    }
}

public class Stacks{
    public static void main(String args[]){
        Stack stackObj = new Stack();
        stackObj.push("Arjun");
        stackObj.push("Tom");

        System.out.println(stackObj.peek());
        System.out.println(stackObj.count());

        stackObj.push("David");
        stackObj.push("Jim");

        System.out.println(stackObj.peek());
        System.out.println(stackObj.count());

        stackObj.pop();

        System.out.println(stackObj.peek());
        System.out.println(stackObj.count());

        stackObj.push("Jim");
        stackObj.push("Eric");
        stackObj.push("Terrance");
    }
}